assignments/array-methods.js

```diff
@@ -56,28 +56,84 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 let fullName = [];
+
+runners.forEach((runner) => {
+    fullName.push(`${runner.first_name} ${runner.last_name}`);
+})
+
 console.log(fullName);
 
+
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+
+runners.map((runner) => {
+    allCaps.push(runner.first_name.toUpperCase());
+})
+
 console.log(allCaps); 
 
+
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+
+runners.filter((runner) => {
+    if (runner.shirt_size === "L") {
+        largeShirts.push(`${runner.first_name} ${runner.last_name}`);
+    }
+})
+
 console.log(largeShirts);
 
+// Alternative solution just for fun
+
+const diffSize = runners.filter((runner) => {
+    if (runner.shirt_size === "L") return runner;
+}).map((runner) => {
+    return `${runner.first_name} ${runner.last_name}`;
+})
+
+console.log(diffSize);
+
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce(function(value, currentEntry) {
+    return value + currentEntry.donation;
+}, 0);
+
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 - Find the names of the companies who donated more than $250 and sort them alphabetically for Platinum Sponsorship plaque
+
+const platinumSpon = [];
+
+runners.filter((company) => {
+    if (company.donation > 250) {
+        platinumSpon.push(company.company_name);
+    }
+});
+
+console.log(platinumSpon.sort());
+
+// Problem 2 - Generate list of runners and their shirt size for the t-shirt distribution team
+
+const shirtSizeList = runners.map((runner) => {
+    return `${runner.last_name}, ${runner.first_name}: ${runner.shirt_size}`;
+})
+
+console.log(shirtSizeList.sort().join("\n"));
+
+// Problem 3 - List of companies and donation amount for tax write off receipts
+
+let deductionReceipts = [];
 
-// Problem 2
+runners.forEach((company) => {
+    deductionReceipts.push({company: `${company.company_name}`, amount: `$${company.donation}.00`});
+})
 
-// Problem 3
\ No newline at end of file
+console.log(deductionReceipts);
```

Comment:
Excellent work! :1st_place_medal:

Reply:
👍it was fun to think of different situations